### PR TITLE
Changes necessary to support jsonschema 3.x

### PR DESCRIPTION
**Pull request by eslavich**
It's been [pointed out](https://github.com/spacetelescope/asdf/issues/530#issuecomment-503811670) that jsonschema 2.6 is incompatible with Python 3.8, so at some point we'll need to upgrade.  These changes make jwst compatible with both jsonschema 2.6 and 3.0.

### CODE REVIEW

File: jwst/datamodels/fits_support.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ def _fits_item_recurse(validator, items, instance, schema):
 def _fits_type(validator, items, instance, schema):
     if instance in ('N/A', '#TODO', '', None):
         return
-    return validators._validators.type_draft4(validator, items, instance, schema)
```
Comment (eslavich) [inline jwst/datamodels/fits_support.py]: This method disappears in jsonschema 3.0, but luckily there is a way to access it through a public API that does not change between the versions.

File: jwst/associations/lib/asn_schema_jw_level2b.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "$schema": "http://json-schema.org/schema#",
```
Comment (eslavich) [inline jwst/associations/lib/asn_schema_jw_level2b.json]: This causes jsonschema to use the latest draft, whatever that is, which is a deprecated behavior in jsonschema 3.0.  Also, the latest draft is now draft-07, whereas in jsonschema 2.6 it was draft-04 anyway.

File: jwst/associations/lib/asn_schema_jw_level2b.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
...
+    "$schema": "http://json-schema.org/draft-04/schema",
```
Comment (hbushouse) [inline jwst/associations/lib/asn_schema_jw_level2b.json]: So we'll have to manually update this whenever we want to start using a new draft?

Comment (eslavich) [reply jwst/associations/lib/asn_schema_jw_level2b.json]: Yes that's correct.  Incidentally, later versions aren't backwards compatible with draft-04, so we'll need to audit these files anyway.

File: jwst/associations/lib/asn_schema_jw_level2b.json (hunk 2 above)
Comment (nden) [reply jwst/associations/lib/asn_schema_jw_level2b.json]: ping @stscieisenhamer

**Comment by nden** [comment]
don't forget the change log

**Comment by nden** [comment]
@eslavich Could you fix the conflict?

**Pull request merged**